source/commands/evergreen/takedown.js

```diff
@@ -26,14 +26,13 @@ async function executeSubcommand(interaction, database, runMode, ...[logicLayer]
 		withResponse: true
 	}).then(response => response.resource.message.awaitMessageComponent({ time: 120000, componentType: ComponentType.StringSelect })).then(async collectedInteraction => {
 		const [bountyId] = collectedInteraction.values;
-		const bounty = await database.models.Bounty.findByPk(bountyId, { include: database.models.Bounty.Company });
+		const [bounty] = openBounties.splice(openBounties.findIndex(bounty => bounty.id === bountyId), 1);
```

Comment:
Using `splice` here feels weird, to be honest. I think `filter` might be a better solution, since you would be able to roll the `findIndex` functionality into the first function and not need to splice exactly 1 result. Since the `findIndex` function searches by ID, it is also reasonable that we only get 1 result out of this method.

Reply:
As part of "one of frontend responsibilities is managing the resource store", the side-effect of removing the bounty element from `openBounties` is utilized later on.

Reply:
Again, I don't think this pattern is bad, per se, I just think that `filter` does the job here better than `splice` and is more direct about the purpose of the code being executed here. (We're filtering down the original list to just the element with the given ID.)

Reply:
Doesn't `filter` not give you reference to the removed stuff? We need that as well to tell the database to `destroy()`.

Reply:
```suggestion
		const [bounty] = openBounties.filter(bounty => bounty.id === bountyId);
```
As much as this suggestion doesn't _guarantee_ an array with a single element, the database should be enforcing that there's only one bounty with each ID.

Reply:
Oh, huh. I'd probably use `.find()` for that specific case since it ends on match. Even then though, `.filter()` doesn't change the original array, so the later `openBounties` checks would need to account for that.

To some extent, this will get solved automatically when we make the `deleteBounty()` endpoint, as then we won't need references to both the bounty to be removed and the remaining bounties in the frontend.

Reply:
Yeah, `find` would also do the trick in the same way, but if we ever got paranoid of duplicate IDs, `filter` would give us an opportunity to get the duplicates. (We likely don't need to and shouldn't need to worry about this, though.)

I'm not a fan of needing to depend on the side effect of `splice` here, though. I definitely didn't catch that, since I'm not used to code needing that side effect, and if they do need that side effect, they overwrite the original variable in the process to make that intent clear. This also makes a bit of the later code confusing, since `openBounties` _should_ only be one element at that point (and we should only have 1 bounty per bounty ID). If that's the case, why not just use `bounty` in those cases, instead?

Reply:
I guess the clarity issues are real. `openBounties` at that point is all evergreen bounties except the one about to be deleted.

Reply:
Ah, I forgot entirely about that feature of `splice`, which _does_ make it reasonable to use here. I'm still not enthusiastic about the side effects usage, but it _is_ useful for the critical path of the code.